staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go

```diff
@@ -306,12 +274,13 @@ func TestErrConnKilled(t *testing.T) {
 	ts := httptest.NewServer(WithPanicRecovery(handler, resolver))
 	defer ts.Close()
 
-	_, err = http.Get(ts.URL)
+	_, err := http.Get(ts.URL)
 	if err == nil {
 		t.Fatal("expected to receive an error")
 	}
 
-	capturedOutput := readStdErr()
+	klog.Flush()
+	capturedOutput := buf.String()
```

Comment:
```suggestion
	klog.Flush()
	klog.SetBuffer(&bytes.Buffer{}) // prevent further writes into buf
	capturedOutput := buf.String()
```

Reply:
It seems like klog has no `SetBuffer` function.